ap.md

```diff
@@ -0,0 +1,256 @@
+
+{% highlight html %}
+<!DOCTYPE html>
+<html lang="en">
+<head>
+  <meta charset="UTF-8">
+  <title>Tessel Web App</title>
+</head>
+<body>
+  <h1>Hello from Tessel!</h1>
+  <p>Press a button to toggle an LED.</p>
+  <ul>
+    <li>
+      <button class="led-button" data-led="2">Green</button>
+      Status: <span class="led-status"></span>
+    </li>
+    <li>
+      <button class="led-button" data-led="3">Blue</button>
+      Status: <span class="led-status"></span>
+    </li>
+  </ul>
+</body>
+</html>
+{% endhighlight %}
+
+<div class="small-12 large-6 columns">
+In that html, we have a heading, a line of instructions for the user, and a list of buttons for controlling a corresponding LED on the Tessel. Below that list of buttons we'll add some JavaScript for communicating with the Tessel:
+</div>
+
+<div class="small-12 large-6 columns">
+![web app screenshot](./images/ap-web-app-preview.png)
+</div>
+
+We will add the following JavaScript before `</body>` tag:
+
+{% highlight html %}
+<script type="text/javascript">
+  // Get a NodeList of elements with the class 'led-button'
+  var buttons = document.querySelectorAll('.led-button');
+
+  // Iterate through that Nodelist and add a 'click' EventListener
+  Array.prototype.forEach.call(buttons, function (button) {
+    button.addEventListener('click', toggleLed);
+  });
+
+  // Our event handler function for 'click' event on the LED buttons
+  function toggleLed (event) {
+    var button = event.target;
+    var ledIndex = button.getAttribute('data-led'); // The index of the led in the Tessel.led array
+    var statusNode = button.parentNode.querySelector('.led-status'); // The sibling status <span> to update
+
+    // Create a new XHR for communicating requests to our Tessel server
+    var req = new XMLHttpRequest();
+
+    // Open a GET request to '/leds/:index'
+    req.open('GET', '/leds/' + ledIndex);
+
+    // Once the request gets a successful response, update that statusNode with the status of the LED.
+    req.onload = function(e) {
+      if (req.readyState == 4 && req.status == 200) {
+        var response = JSON.parse(req.responseText);
+        statusNode.textContent = response.on ? 'ON' : 'OFF';
+      } else { 
+        console.log('Error', e); // If something went wrong, log that event to the console.
+      }
+    }
+    req.send(); // Send our request to the server
+  }
+</script>
+{% endhighlight %}
+
+Now let's check out `ap.js` again to finish up the project. First, we're going to make a few changes to our server setup:
+
+{% highlight javascript %}
+// These two dependencies remain the same
+var tessel = require('tessel');
+var http = require('http');
+
+// Require two other core Node.js modules
+var fs = require('fs');
+var url = require('url');
+
+var server = http.createServer(function (request, response) {
+  // Here is the biggest change
```

Comment:
// Here is the biggest change: handle LED requests
(or something more purposive)